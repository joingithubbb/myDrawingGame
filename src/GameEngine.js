import React, { useReducer } from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";
import { Switch } from "react-router";
import { Game } from "./Game";
import { Home } from "./Home";
import { End } from "./End";
import * as tf from "@tensorflow/tfjs";

const model = tf.loadModel("./model/model.json");

// REQUIREMENT: Use of Context to avoid "prop-drilling"
const PointsContext = React.createContext();

function transferPointsReducer(points, action) {
  switch (action.transferredPoint) {
    case "increment":
      return points + 1;
    case "decrement":
      return points - 1;
    default:
  }
}

function GameEngine() {

  const initialState = 0;

  // REQUIREMENT: Appropriate use of the useReducer() hook
  var [points, setPoints] = useReducer(transferPointsReducer, initialState);

  function setThePoints(increase) {
    if (increase === true) {
      setPoints({ transferredPoint: "increment" });
    }

    else {
      setPoints({ transferredPoint: "decrement" });
    }

  }

  // REQUIREMENT: Use of Context to avoid "prop-drilling"
  return (
    <PointsContext.Provider value={{
      "points": points,
      "setThePoints": setThePoints
    }}>
      <AppRouter setThePoints={setThePoints} />
    </PointsContext.Provider>
  );
}

function HomeRouter() {
  return <Home />;
}

function GameRouter(props) {

  return (
    <Game setThePoints={props.setThePoints} model={model} />
  );
}

function EndRouter() {

  return (
    <End />
  );
}

function NoMatch() {

  return <h1>Nothing to see here - Four Oh! Four</h1>;
}

function AppRouter(props) {
  return (
    <Router>
      <div>
        <Switch>
          <Route path="/" exact component={HomeRouter} />
          <Route
            path="/game"
            render={(routeProps) => (<GameRouter {...routeProps} {...props}
              setThePoints={props.setThePoints} />)}
          />
          <Route path="/end" component={EndRouter} />
          <Route component={NoMatch} />
        </Switch>
      </div>
    </Router>
  );
}

export { GameEngine };
export const PointsProvider = PointsContext.Provider;
export const PointsConsumer = PointsContext.Consumer;
export { PointsContext };
export { HomeRouter };
export { GameRouter };
export { EndRouter };