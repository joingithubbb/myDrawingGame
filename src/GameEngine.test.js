import React from 'react';
import ReactDOM from 'react-dom';
import { GameEngine } from './GameEngine';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from "react-test-renderer";
Enzyme.configure({ adapter: new Adapter() });

// REQUIREMENT: Test suite that test the features of your UI

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GameEngine />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

//For checking after a refactor
it("should render <App />", () => {
    const componentTree = renderer.create(<GameEngine />).toJSON();
    expect(componentTree).toMatchSnapshot();
  });