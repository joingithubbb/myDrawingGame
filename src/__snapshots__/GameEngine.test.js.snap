// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <App /> 1`] = `
<div>
  <div>
    <h1>
      Sketch
    </h1>
    <div>
      <h2>
        Welcome to Sketch the Drawing Game!
      </h2>
      <h3>
        This is a game inspired by Google's "Quick, Draw!" - You'll have to draw quickly!
      </h3>
      <br />
    </div>
    <a
      href="/game"
      onClick={[Function]}
    >
      Proceed
    </a>
  </div>
</div>
`;
